src/flash/display/Loader.ts

```diff
@@ -83,77 +82,98 @@ module Shumway.AVM2.AS.flash.display {
     };
     static initializer: any = function() {
       var self: Loader = this;
-      DisplayObject._advancableInstances.push(self);
+      display.DisplayObject._advancableInstances.push(self);
     };
 
     static classSymbols: string [] = null;
     static instanceSymbols: string [] = null;
 
-
     /**
-     * Handles the load status and dispatches progress events. This gets manually triggered in the
-     * event loop to ensure the correct order of operations.
+     * In each turn of the event loop, Loader events are processed in two batches:
+     * first INIT and COMPLETE events are dispatched for all active Loaders, then
+     * OPEN and PROGRESS.
+     *
+     * A slightly weird result of this is that INIT and COMPLETE are dispatched one
+     * turn *later* than the other events: INIT is only dispatched if the loadStatus
```

Comment:
This is not necessarily related. It is true that INIT events (and therefore COMPLETE as well, since it depends on INIT being fired first) are dispatches AT LEAST one turn later, since INIT has to be fired AFTER the first frame is completely loaded and all its initial children are constructed (which, depending on download speed, can take longer than a single event loop turn). Again, a DisplayObjectFlags.FirstFrameReady or DisplayObjectFlags.FirstFrameConstructed flag could help here to decide whether to dispatch INIT.

It also looks like that processing loading events for the root loader takes one complete turn exclusively, meaning all loaders created on the very first frame are getting processed starting from frame 2. That really only is the case on the very first turn of the event loop. All loaders created in frames after that, are processed within the same turn.